packages/bsky/src/notifications.ts

```diff
@@ -0,0 +1,285 @@
+      recordUri,
+    } = notif
+    // 1. Get author's display name
+    const displayName = await this.db.db
+      .selectFrom('profile')
+      .where('creator', '=', authorDid)
+      .select(['displayName'])
+      .executeTakeFirst()
+    if (!displayName || !displayName?.displayName) {
+      throw new Error('Failed to get display name. User has no profile')
+    }
+    const author = NotificationServer.sanitizeDisplayName(
+      displayName.displayName,
+    )
+
+    // 2. Get post data content
+    // if reply, quote, or mention, get URI of the postRecord
+    // if like, or custom feed like, or repost get the URI of the reasonSubject
+    // if follow, get the URI of the author's profile
+    let title = ''
+    let body = ''
+
+    // check follow first and mention first because they don't have subjectUri and return
+    if (reason === 'follow') {
+      title = 'New follower!'
+      body = `${author} has followed you`
+      return { title, body }
+    } else if (reason === 'mention') {
+      title = `${author} mentioned you`
+      const mentionedPostData = await this.db.db
+        .selectFrom('post')
+        .where('uri', '=', recordUri)
+        .selectAll()
+        .executeTakeFirst()
+      body = mentionedPostData?.text || ''
+      return { title, body }
+    }
+
+    if (!subjectUri) {
+      throw new Error('Failed to get subject URI')
+    }
+
+    const postData = await this.db.db
+      .selectFrom('post')
+      .where('uri', '=', subjectUri)
+      .selectAll()
+      .executeTakeFirst()
+    if (!postData) {
+      throw new Error('Failed to get post data')
+    }
+
+    if (reason === 'like') {
+      title = `${author} liked your post`
+      body = postData?.text || ''
+      // custom feed like
+      if (subjectUri?.includes('feed.generator')) {
+        title = `${author} liked your custom feed`
+        body = `${new AtUri(subjectUri).rkey}`
+      }
+    } else if (reason === 'reply') {
+      title = `${author} replied to your post`
+      body = postData?.text || ''
+    } else if (reason === 'quote') {
+      title = `${author} quoted your post`
+      body = postData?.text || ''
+    } else if (reason === 'repost') {
+      title = `${author} reposted your post`
+      body = postData?.text || ''
+
+      // TODO: handle images
+      // let image
+      // if (
+      //   AppBskyEmbedImages.isView(
+      //     notification.additionalPost?.thread?.post.embed,
+      //   ) &&
+      //   notification.additionalPost?.thread?.post.embed.images[0]?.thumb
+      // ) {
+      //   image = notification.additionalPost.thread.post.embed.images[0].thumb
+      // }
+    }
+
+    if (title === '' && body === '') {
+      throw new Error('Failed to get title and body')
+    }
+
+    return { title: title, body: body }
+  }
+
+  static sanitizeDisplayName(str: string): string {
```

Comment:
how necessary do you think this is? genuinely asking. I don't love it being in the server since we don't do much other sanitization :/ but also push notifications are kinda like "client in the server" lol

Reply:
We do this sanitizing on the client so that is why I moved it to the sever. I think it is for people who tried to "act" as verified by putting a checkmark in their name. Upto you and @pfrazee if we should remove it.

Reply:
I can probably remove it since it doesn't seem like people are doing that anymore but I don't know if its still an issue

Reply:
I think it's fine to remove tbh - when they open the app, they'd see the sanitized name

Reply:
Removed